app/models/form_owner_settings.rb

```diff
@@ -21,16 +21,19 @@ def update
     version = MetadataApiClient::Version.create(service_id:, payload: new_metadata)
     if version.errors?
       errors.add(:base, :invalid, message: version.errors)
-      false
+      return false
     else
       @version = version
     end
+    true
   end
 
   private
 
   def email_exists?
-    @new_user_id = User.all.map { |user| user.id if user.email == @form_owner }.compact.first
+    @new_user_id = User.all.map { |user|
```

Comment:
It probably won't make a noticeable performance difference but this will map over and call a regex match on every row in the user table, I think it would be more performant to offload the match onto the database with something like `User.where('lower(email) = ?', @form_owner.downcase).first`

I don't think that the User table is ever going to be big enough that this is an issue though

Reply:
I was about to say the same, and then sadly realised the user email is encrypted so we can't run DB queries 😢 

But I think this alternative might be a bit more performant and easier to read?

```ruby
User.all.detect { |u| u.email.downcase == @form_owner.downcase }.present?
```